src/ypd/model/user.py

```diff
@@ -214,3 +219,62 @@ def get_by_id(cls, id, session=None):
             session (Session): session to perform the query on. Supplied by decorator
         """
         return session.query(User).filter_by(id=id).one_or_none()
+    
+    @classmethod
+    @SessionManager.with_session
+    def get_by_username(cls, username, session=None):
+        """Gets the User object with the unique username
+        
+        Args:
+            username (str): username of User object to get
+
+        Kwargs:
+            session (Session): session to perform the query on. Supplied by decorator
+        """
+        return session.query(User).filter_by(username=username).one_or_none()
+    
+    @classmethod
+    def password_check(self, password, confirm_password):
+        """Takes in user form password and confirmed password, and validates
+        Args:
+            password (str): Password to be stored in user
+            confirm_password (str): Password for comparing to for validation
+        Raises:    
+            ValueError: If password and confirm_password length and elements do no match
+        Returns:
+            password (hash): Password hash to be stored in account 
+        """
+        if confirm_password != password:
+            raise ValueError('New password and cofirm password do not match!!!')
+        elif len(confirm_password) < 8  or len(password) < 8:
+            raise ValueError('Password must be at least 8 characters long!')
+        else:
+            return generate_password_hash(password)
+
+    @SessionManager.with_session
+    def update_password(self, password, confirm_password, session=None):
+        """Updates current user object password with new password
+        
+        Args:
+            password (str): password of User object to get
+            confirm_password (str): password of User object to get
+
+        Kwargs:
+            session (Session): session to perform the query on. Supplied by decorator
+
+        Raises Value
+        """
+        self.password = User.password_check(password, confirm_password)
+        session.add(self)    
+    
+    # @classmethod
+    @SessionManager.with_session
+    def delete_account(self, session=None):
+        
+        acc = User.get_by_username(self.username)
```

Comment:
Is this supposed to be a classmethod? Why query for acc if we already have self?

Reply:
It throws errors if I do it that way